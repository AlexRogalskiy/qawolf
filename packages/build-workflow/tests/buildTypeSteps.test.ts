import { loadEvents } from "@qawolf/fixtures";
import { buildTypeSteps } from "../";

describe("buildTypeSteps", () => {
  it("builds expected steps for login", async () => {
    const events = await loadEvents("scroll_login");
    const steps = buildTypeSteps(events);
    expect(steps).toMatchSnapshot();
  });

  it("builds expected steps for hotkeys", async () => {
    const events = await loadEvents("githubIssuesShortcut");
    const steps = buildTypeSteps(events);
    expect(steps).toMatchSnapshot();
  });
});
