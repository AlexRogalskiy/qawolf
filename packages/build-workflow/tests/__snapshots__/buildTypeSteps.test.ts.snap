// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildTypeSteps builds expected steps 1`] = `
Array [
  Object {
    "action": "type",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {
            "class": "large-6 small-12 columns",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {
            "class": "row",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "id": "username",
          "labels": "Username",
          "name": "username",
          "type": "text",
        },
        "children": Array [],
        "name": "input",
        "type": "tag",
        "voidElement": true,
      },
    },
    "index": 26,
    "page": 0,
    "value": "↓KeyT↑KeyT↓KeyO↑KeyO↓KeyM↑KeyM↓KeyS↑KeyS↓KeyM↑KeyM↓KeyI↑KeyI↓KeyT↑KeyT↓KeyH↑KeyH",
  },
  Object {
    "action": "type",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {
            "class": "large-6 small-12 columns",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {
            "class": "row",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "id": "username",
          "labels": "Username",
          "name": "username",
          "type": "text",
        },
        "children": Array [],
        "name": "input",
        "type": "tag",
        "voidElement": true,
      },
    },
    "index": 42,
    "page": 0,
    "value": "↓Tab↑Tab",
  },
  Object {
    "action": "type",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {
            "class": "large-6 small-12 columns",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {
            "class": "row",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "id": "password",
          "labels": "Password",
          "name": "password",
          "type": "password",
        },
        "children": Array [],
        "name": "input",
        "type": "tag",
        "voidElement": true,
      },
    },
    "index": 46,
    "page": 0,
    "value": "SuperSecretPassword!",
  },
  Object {
    "action": "type",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {
            "class": "large-6 small-12 columns",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {
            "class": "row",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "id": "password",
          "labels": "Password",
          "name": "password",
          "type": "password",
        },
        "children": Array [],
        "name": "input",
        "type": "tag",
        "voidElement": true,
      },
    },
    "index": 49,
    "page": 0,
    "value": "↓Enter↑Enter",
  },
]
`;
